Error while building

I think you need to add `@wordpress/lazy-import` to `packages/is-shallow-equal/package.json`

```
 node packages/is-shallow-equal/benchmark/index.js
(node:10474) UnhandledPromiseRejectionWarning: Error: Cannot find module '@wordpress/lazy-import.5ab856344c555e50779f56b4cd1f2e02'
Require stack:
- /home/mkaz/src/wp/gutenberg/packages/lazy-import/lib/index.js
- /home/mkaz/src/wp/gutenberg/packages/is-shallow-equal/benchmark/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:831:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:903:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at lazyImport (/home/mkaz/src/wp/gutenberg/packages/lazy-import/lib/index.js:138:9)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Promise.all (index 1)
```

--- a/packages/is-shallow-equal/benchmark/index.js
+++ b/packages/is-shallow-equal/benchmark/index.js
@@ -26,23 +26,8 @@ Promise.all( [
 	lazyImport( 'shallow-equal@^1.2.1' ),
 	lazyImport( 'is-equal-shallow@^0.1.3' ),
 	lazyImport( 'shallow-equals@^1.0.0' ),
-	new Promise( async ( resolve ) => {
-		try {
-			await lazyImport( 'fbjs@^1.0.0' );
-		} catch ( error ) {
-			// The fjbs package throws an error when imported directly. Since
-			// lazyImport will automatically require the module for the resolved
-			// value, anticipate and disregard the error, as long as it's the
-			// expected error message.
-			if (
-				'The fbjs package should not be required without a full path.' !==
-				error.message
-			) {
-				throw error;
-			}
-		}
-
-		resolve( require( 'fbjs/lib/shallowEqual' ) );
+	lazyImport( 'fbjs@^1.0.0', {
+		localPath: './lib/shallowEqual',
 	} ),
 ] ).then(
 	( [

--- a/packages/lazy-import/CHANGELOG.md
+++ b/packages/lazy-import/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### New Feature
+
+-   Allow local paths as an option when trying to import a specific file from the package ([#23751](https://github.com/WordPress/gutenberg/pull/23751)).
+
 ## 1.0.0 (2020-06-15)
 
-- Initial release.
+-   Initial release.

--- a/packages/lazy-import/README.md
+++ b/packages/lazy-import/README.md
@@ -52,7 +52,10 @@ function onInstall() {
 	console.log( 'Installing…' );
 }
 
-lazyImport( 'is-equal-shallow@^0.1.3', { onInstall } ).then( /* ... */ );
+lazyImport( 'fbjs@^1.0.0', {
+	localPath: './lib/shallowEqual',
+	onInstall,
+} ).then(/* ... */);
 ```
 
 Note that `lazyImport` can throw an error when offline and unable to install the dependency using NPM. You may want to anticipate this and provide remediation steps for a failed install, such as logging a warning messsage:
@@ -69,10 +72,17 @@ try {
 
 ### Options
 
+#### `localPath`
+
+-   Type: `string`
+-   Required: No
+
+Local path pointing to a file or directory that can be used when other script that `main` needs to be imported.
+
 #### `onInstall`
 
-- Type: `Function`
-- Required: No
+-   Type: `Function`
+-   Required: No
 
 Function to call if and when the module is being installed. Since installation can cause a delay in script execution, this can be useful to output logging information or display a spinner.
 

--- a/packages/lazy-import/lib/index.js
+++ b/packages/lazy-import/lib/index.js
@@ -10,7 +10,8 @@ const { createHash } = require( 'crypto' );
 /**
  * @typedef WPLazyImportOptions
  *
- * @property {()=>void} onInstall Callback to invoke when install starts.
+ * @property {string}   [localPath] Path to the local directory or file.
+ * @property {()=>void} [onInstall] Callback to invoke when install starts.
  */
 
 /**
@@ -63,6 +64,7 @@ async function install( arg, alias ) {
  * @return {Promise<NodeRequire>} Promise resolving to required module.
  */
 async function lazyImport( arg, options = {} ) {
+	const { localPath = '' } = options;
 	const { rawSpec, name } = npmPackageArg( arg );
 
 	if ( ! name ) {
@@ -77,7 +79,9 @@ async function lazyImport( arg, options = {} ) {
 	// need to verify both availability and version. Version isn't necessary to
 	// account for in this first attempt.
 	try {
-		return require( localModule );
+		if ( require.resolve( localModule ) ) {
+			return require( join( localModule, localPath ) );
+		}
 	} catch ( error ) {
 		if ( error.code !== 'MODULE_NOT_FOUND' ) {
 			throw error;
@@ -85,13 +89,11 @@ async function lazyImport( arg, options = {} ) {
 	}
 
 	try {
-		const resolved = require( name );
-
 		const { version } = require( join( name, 'package.json' ) );
 		if ( semver.satisfies( version, rawSpec ) ) {
 			// Only return with the resolved module if the version is valid per
 			// the parsed arg. Otherwise, fall through to install stage.
-			return resolved;
+			return require( join( name, localPath ) );
 		}
 	} catch ( error ) {
 		if ( error.code !== 'MODULE_NOT_FOUND' ) {
@@ -133,7 +135,7 @@ async function lazyImport( arg, options = {} ) {
 	//
 	// See: https://github.com/WordPress/gutenberg/pull/22684#discussion_r434583858
 
-	return require( localModule );
+	return require( join( localModule, localPath ) );
 }
 
 module.exports = lazyImport;

